pandas/_libs/index.pyx

```diff
@@ -275,6 +275,12 @@ cdef class IndexEngine:
 
         self._ensure_mapping_populated()
         values = np.array(self._get_index_values(), copy=False)
+        targets_mask = isnaobj(targets)
+        values_mask = isnaobj(values)
+        if targets_mask.any():
+            targets[targets_mask] = 0
+            values[values_mask] = 0
```

Comment:
why are you setting to 0? these are indexers so shouldn't these be -1 ?

Reply:
agreed, changed to -1